dlt/extract/incremental/transform.py

```diff
@@ -117,18 +120,49 @@ def find_cursor_value(self, row: TDataItem) -> Any:
 
         Will use compiled JSONPath if present, otherwise it reverts to column search if row is dict
         """
-        row_value: Any = None
         if self._compiled_cursor_path:
-            row_values = find_values(self._compiled_cursor_path, row)
-            if row_values:
-                row_value = row_values[0]
+            cursor_values = find_values(self._compiled_cursor_path, row)
+            if cursor_values == []:
+                if self.on_cursor_value_missing == "raise":
+                    raise IncrementalCursorPathMissing(self.resource_name, self.cursor_path, row)
+                elif self.on_cursor_value_missing == "exclude":
+                    return None
+            elif None in cursor_values:
+                if self.on_cursor_value_missing == "raise":
+                    raise IncrementalCursorPathHasValueNone(
+                        self.resource_name, self.cursor_path, row
+                    )
+                elif self.on_cursor_value_missing == "exclude":
+                    return None
+
+            # ignores the other found values, e.g. when the path is $data.items[*].created_at
+            row_value = cursor_values[0]
         else:
+            row_value = None
             try:
-                row_value = row[self.cursor_path]
-            except Exception:
+                row_value = row.get(self.cursor_path)
+            except AttributeError:
+                # just to pass tests/pipeline/test_pipeline_extra.py::test_dump_trace_freeze_exception
```

Comment:
OK from my side to try getting with `getattr` on top of `get`. Then we support Pydantic models as well. You can leave failing trace test to me. This is testing some kind of super edge case when using pickle on local object. I can mess around to recreate it